src/iris/api.py

```diff
@@ -1038,7 +1042,13 @@ def debug_auth(self, req, resp, resource, params):
             return
 
     def process_resource(self, req, resp, resource, params):  # pragma: no cover
-        req.context['username'] = req.env.get('beaker.session', {}).get('user', None)
+        sso_username = self.sso_manager.authenticate(req)
```

Comment:
Is this going to call the (expensive?) authenticate function on every page request? 

If so should this instead only call authenticate only once per session (upon login) and store the auth's username in the beaker session for fast access?

Reply:
The check itself should be very fast. But more importantly I don't think we want to create a beaker session based off SSO login, I think we would want that login to work only when SSO auth is still fresh and since SSO does not have an active logout flow that pushes to our system we have no way of verifying that except going through the auth flow. Otherwise we can end up in an unexpected state of a user remaining logged in to Iris while logged out of SSO which I think we want to avoid.

Reply:
Makes sense, thanks for the clarification!